web/src/recent_topics_ui.js

```diff
@@ -541,6 +541,12 @@ export function filters_should_hide_topic(topic_data) {
     }
 
     if (!filters.has("include_muted") && topic_data.type === "stream") {
+        // We want to show the unmuted topics within muted streams in the recent topics.
+        const topic_unmuted = Boolean(user_topics.is_topic_unmuted(msg.stream_id, msg.topic));
+        if (topic_unmuted) {
+            return false;
+        }
```

Comment:
Since we don't want to skip the topic search filter below for this, let's just add `& !topic_unmuted` in the condition below.